src/main/java/io/wisetime/idea/branch/BranchHelper.java

```diff
@@ -24,12 +26,19 @@ public String getCurrentBranchName() {
 
   public void onBranchChanged(String branchName) {
     currentBranchName = branchName;
-    updateFrameTitle();
+    ApplicationManager.getApplication().invokeLater(this::updateFrameTitle);
   }
 
   private void updateFrameTitle() {
-    final String projectTitle = ServiceManager.getService(FrameTitleBuilder.class)
-        .getProjectTitle(project);
+    final String projectTitle = FrameTitleBuilder.getInstance().getProjectTitle(project);
+    SettingsState settings = SettingsState.getInstance();
+    if (settings.appendToProjectName) {
+      updateProjectTitle(projectTitle);
```

Comment:
We cannot rollback project name reliably - therefore if settings disabled, leave as is

Reply:
SGTM